modules/oas-page-builder/src/services/database.ts

```diff
@@ -0,0 +1,47 @@
+import { Db, MongoClient } from 'mongodb';
+import { OASFile, OASFileGitHash } from './models/OASFile';
+
+const DB_NAME = process.env.DB_NAME || 'snooty_dotcomprd';
+const COLLECTION_NAME = 'oas_files';
+
+const getAtlasURL = () => {
+  const isHostLocal = process.env.DB_HOST?.includes('localhost');
+  if (isHostLocal) {
+    return `mongodb://${process.env.DB_HOST}/?retryWrites=true&w=majority`;
+  }
+  return `mongodb+srv://${process.env.DB_USER}:${process.env.DB_PASSWORD}@${process.env.DB_HOST}/?retryWrites=true&w=majority`;
+};
+
+const atlasURL = getAtlasURL();
+const client = new MongoClient(atlasURL);
+// cached db object, so we can handle initial connection process once if unitialized
+let dbInstance: Db;
+
+// Handles memoization of db object, and initial connection logic if needs to be initialized
+const db = async () => {
+  if (!dbInstance) {
+    try {
+      await client.connect();
+      dbInstance = client.db(DB_NAME);
+    } catch (error) {
+      console.error(`Error at db client connection: ${error}`);
+      throw error;
+    }
+  }
+  return dbInstance;
+};
```

Comment:
This logic is the same as the persistence module's. I wanted a way to sort of share the logic between the two, but might be difficult if we wanted to treat the 2 as independent modules

Reply:
🤔 it might make sense to manage this dependency in a shared lib in the future, but I think this is sufficient for now. We can always revisit if we add a 3rd module and feel that the logic is starting to get copy-pasted. 😄